src/main/assisted-installer-controller/assisted_installer_main.go

```diff
@@ -250,8 +250,32 @@ func waitForInstallation(client inventory_client.InventoryClient, log logrus.Fie
 // the inventory client because assisted-service is deployed on the bootstrap node
 // and when the bootstrap node reboots to join the cluster, assisted-service becomes
 // unavailable.
-func waitForInstallationAgentBasedInstaller(kubeClient k8s_client.K8SClient, log logrus.FieldLogger) {
+func waitForInstallationAgentBasedInstaller(kubeClient k8s_client.K8SClient, log logrus.FieldLogger, removeUninitialiedTaint bool) {
 	for {
+		if removeUninitialiedTaint {
+			// After the boostrap node has rebooted, assisted-service becomes unavailable
+			// and nodes can no longer be untainted through
+			// assistedController.WaitAndUpdateNodesStatus(mainContext, &wg, removeUninitializedTaint).
+			// because the call to retrieve hosts (c.ic.GetHosts(ctxReq, log, ignoreStatuses))
+			// will fail immediately.
+			// Here we have an alternative mechanism to untaint nodes when
+			// assisted-service is no longer available.
+			nodes, err := kubeClient.ListNodes()
+			if err != nil {
+				log.WithError(err).Error("Failed to get list of nodes from k8s client")
+			}
+			for _, node := range nodes.Items {
+				if taintIndex, err := kubeClient.HasUninitializedTaint(node.Name); taintIndex != -1 {
+					if err != nil {
+						log.WithError(err).Errorf("Checking if node %s contained uninitialized taint failed", node.Name)
```

Comment:
Should continue on error here instead of calling UntaintNode.

Reply:
Thanks. A continue should be added here on error.